wsidicomizer/openslide.py

```diff
@@ -377,9 +432,16 @@ def _get_encoded_tile(
         bytes
             Tile bytes.
         """
-        if z not in self.focal_planes or path not in self.optical_paths:
-            raise ValueError
-        return self._encode(self._get_tile(tile))
+        if z not in self.focal_planes:
+            raise WsiDicomNotFoundError(f'focal plane {z}', str(self))
+        if z not in self.focal_planes:
```

Comment:
You have the following line twice now. I assume this second one should be about optical paths?
```python
if z not in self.focal_planes:
```

Also think about why this could get passed your test suite. Try making a test that discovers this error before fixing it.

Reply:
Fixed and added test.